apps/internal/oauth/ops/authority/authority.go

```diff
@@ -138,6 +138,38 @@ const (
 	ADFS = "ADFS"
 )
 
+type AuthenticationScheme interface {
+	// Extra parameters that are added to the request to the /token endpoint.
+	TokenRequestParams() map[string]string
+	// Key ID of the public / private key pair used by the encryption algorithm, if any.
+	// Tokens obtained by authentication schemes that use this are bound to the KeyId, i.e.
+	// if a different kid is presented, the access token cannot be used.
+	KeyID() string
+	// Creates the access token that goes into an Authorization HTTP header.
+	FormatAccessToken(accessToken string) (string, error)
```

Comment:
@manojampalam - I liked your idea of being able to modify the entire AuthenticaitonResult object, not just the access token. But this can work as well, not a blocking comment.